### PR TITLE
ES2015: Exception for escaped characters in keywords

### PR DESCRIPTION
**Pull request by hydroper**
Fix https://github.com/jquery/esprima/issues/1612

### CODE REVIEW

File: src/scanner.ts [hunk 1]
```diff
@@ -498,6 +504,12 @@ export class Scanner {
             type = Token.Identifier;
         }
 
+        if (this.escapedChars && type !== Token.Identifier) {
+        	let col: number = ( start - this.lineStart ) + 1;
+            this.errorHandler.throwError(start, this.lineNumber,
+            col, "Keyword must not contain escaped characters");
+        }
```
Comment (ariya) [inline src/scanner.ts]: This block has incorrect coding style.

**Comment by ariya** [comment]
Some tests are still needed, otherwise the coverage will regress.

**Comment by hydroper** [comment]
These tests are still done with npm test? Do I have to use the GitHub app?

**Comment by hydroper** [comment]
I'm not really sure about these tests... I successfully compiled the parser with npm test before I did this pull request.

**Comment by ariya** [comment]
> Do I have to use the GitHub app, too ?

What do you mean by GitHub app?

**Comment by hydroper** [comment]
I mean the GitHub program for Windows/other systems.

**Comment by ariya** [comment]
Something must be still wrong. Look at the Travis CI output (you'll get the same by running `npm test`):
```
> esprima@4.0.0-dev tslint /home/travis/build/jquery/esprima
> tslint src/*.ts
src/scanner.ts[509, 71]: " should be '
```
This is what I already commented earlier (on the coding style) and it's not fixed yet.

File: src/scanner.ts [hunk 2]
```diff
@@ -505,8 +505,8 @@ export class Scanner {
         }
 
         if (this.escapedChars && type !== Token.Identifier) {
-        	let col: number = ( start - this.lineStart ) + 1;
-            this.errorHandler.throwError(start, this.lineNumber, col, "Keyword must not contain escaped characters");
+            let col: number = ( start - this.lineStart ) + 1;
```
Comment (ariya) [inline src/scanner.ts]: Too much spaces around, see the rest of the source file for the example on how to format it.

**Comment by ariya** [comment]
> mean the GitHub program for Windows/other systems.

@hydroper I'm not sure what that is.

**Comment by ariya** [comment]
@hydroper How are you editing this commit? Seems that you push every little changes without actually running and passing `npm test`, that's confusing.

**Comment by hydroper** [comment]
I've downloaded the repository and I run npm test in the root of the repository, then dist\esprima.js gets updated.

**Comment by ariya** [comment]
> Edit: I also receive a lot of exceptions while running npm test, but these look to be related to node.js.

What exceptions? Every PR needs to pass `npm test` cleanly.

**Comment by hydroper** [comment]
The top-level errors:

image

(I think there're more; none related to scanner.ts)

**Comment by ariya** [comment]
@hydroper That was an important error, your setup failed to prepare the environment for the unit tests. Without fixing it, you are not running the tests at all 😞

**Comment by hydroper** [comment]
Sorry, I'm not sure how it works, but you can close this pull request and make a new one including the scanner.ts edits.

**Pull request closed without merging**